index.src.html

```diff
@@ -599,45 +656,59 @@ <h2 id="integrations">Integrations</h2>
 
+  <h3 id="secure-context-restriction">Secure context restriction</h4>
+
+  UAs must not allow <a lt="secure context">non-secure</a>
+  [=IP address space/public=] contexts to request resources from
+  [=private addresses=], even if the private server would opt-in to such a
+  request via a preflight. Making requests to [=IP address space/private=]
+  resources presents risks which are mitigated by ensuring the integrity of the
+  [=request/client=] which initiates the request. In particular, network
+  attackers should not be able to trivially exploit an endpoint's consent to a
+  non-secure origin.
+
+  Mixed content checks [[MIXED-CONTENT-2]] prevent secure contexts from making
+  requests over HTTP, so this restriction would seem to require that private
+  network servers migrate to HTTPS. This is often difficult to impossible. A
+  new [=permission prompt=] is introduced to allow secure contexts to make
+  requests over HTTP to the private network anyway, given user consent.
+
   <h3 id="integration-permissions">Integration with Permissions </h3>
 
   This document defines a [=powerful feature=] identified by the [=powerful
-  feature/name=] <dfn export permission><code>
-  "mixed-content-private-network-access"</code></dfn>". It overrides the
-  following type:
+  feature/name=] <dfn export permission>`"private-network-access"`</dfn>. It
+  overrides the following type:
 
   <dl>
     <dt>[=powerful feature/permission descriptor type=]</dt>
     <dd>
       The [=powerful feature/permission descriptor type=] of the
-      <a permission><code>"mixed-content-private-network-access"</code></a>
-      feature is defined by the following WebIDL interface that
-      [=dictionary/inherits=] from the default [=powerful feature/permission
-      descriptor type=]:
+      <a permission>`"private-network-access"`</a> feature is defined by the
+      following WebIDL interface that [=dictionary/inherits=] from the
+      default [=powerful feature/permission descriptor type=]:
       <pre class="idl">
-        dictionary MixedContentPrivateNetworkAccessPermissionDescriptor
+        dictionary PrivateNetworkAccessPermissionDescriptor
             : PermissionDescriptor {
           DOMString id;
         };
       </pre>
     </dd>
-    <dt>[=powerful feature/extra permission data type=]</dt>
   </dl>
 
   <h3 id="integration-mixed-content">Integration with Mixed Content</h3>
   The [=Should fetching request be blocked as mixed content?=] is amended to add
-  the following condition to one of the <strong>allowed</strong> conditions:
+  the following condition to one of the **allowed** conditions:
 
-  1.  |request|'s [=request/URL=]'s [=url/scheme=] is <code>https</code>, and
-      |request|'s [=request/target IP address space=] is <code>private</code> or
-      <code>local</code>.
+  1.  |request|'s [=request/URL=]'s [=url/scheme=] is `http`, and |request|'s
+      [=request/target IP address space=] is `private` or `local`.
```

Comment:
These are strings in the IDL, I believe:

```suggestion
      [=request/target IP address space=] is `"private"` or `"local"`.
```

Reply:
Per your comments below to convert values, these should probably be [=IP address space/private=] and [=IP address space/local=]?